src/server/req_quejob.c

```diff
@@ -1968,29 +1964,33 @@ req_commit(struct batch_request *preq)
 	}
 
 	/* need to format message first, before request goes away */
-
-	(void)snprintf(log_buffer, sizeof(log_buffer), msg_jobnew,
-		preq->rq_user, preq->rq_host,
-		pj->ji_wattr[(int)JOB_ATR_job_owner].at_val.at_str,
-		pj->ji_wattr[(int)JOB_ATR_jobname].at_val.at_str,
-		pj->ji_qhdr->qu_qs.qu_name);
+	pbs_asprintf(&newjobmsg, msg_jobnew,
```

Comment:
If you just use log_eventf on line 1988, you can get rid of newjobmsg

Reply:
We can't because in at this line we are using preq struct, which will get released in a call to reply_jobid() at line 1980. So we have to create a message before we release preq struct.